Composer/packages/client/src/recoilModel/dispatchers/notification.ts

```diff
@@ -53,10 +53,17 @@ export const notificationDispatcher = () => {
     set(notificationsState(id), (notification) => ({ ...notification, hidden: true }));
   });
 
+  const updateNotification = useRecoilCallback(
+    (callbackHelper: CallbackInterface) => (id: string, newValue: CardProps) => {
```

Comment:
newValue type should be ```Partial<CardProps>``` so the called can change it partially.

change updateNotificationInternal accordingly to accept partial props changes:
```typescript
export const updateNotificationInternal = ({ set }: CallbackInterface, id: string, newValue: Partial<CardProps>) => {
// I'm not sure why id is being re-set here?
  set(notificationsState(id), (current) => { ...current, ...newValue, id: id });
  // check if notification exist
  set(notificationIdsState, (notificationIds) => {
    if (notificationIds.some((notificationId) => notificationId === id)) {
      return notificationIds;
    } else {
      return [...notificationIds, id];
    }
  });
};
```

Reply:
This will allow a caller to change notification partially, for example, change the status only by passing ONLY status.

Reply:
Refactor.    `set(notificationsState(id), (current) => ({ ...current, ...newValue }));` . Id should not be reset